homeassistant/components/devialet/media_player.py

```diff
@@ -0,0 +1,274 @@
+"""Support for Devialet Phantom speakers."""
+from __future__ import annotations
+
+import datetime
+from datetime import timedelta
+
+from devialet import DevialetApi
+from devialet.const import NORMAL_INPUTS
+
+from homeassistant.components.media_player import (
+    MediaPlayerEntity,
+    MediaPlayerEntityFeature,
+    MediaPlayerState,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_NAME
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DEFAULT_SCAN_INTERVAL, DOMAIN, MANUFACTURER, SOUND_MODES
+
+SCAN_INTERVAL = timedelta(seconds=DEFAULT_SCAN_INTERVAL)
+
+SUPPORT_DEVIALET = (
+    MediaPlayerEntityFeature.VOLUME_SET
+    | MediaPlayerEntityFeature.VOLUME_MUTE
+    | MediaPlayerEntityFeature.TURN_OFF
+    | MediaPlayerEntityFeature.SELECT_SOURCE
+    | MediaPlayerEntityFeature.SELECT_SOUND_MODE
+)
+
+SUPPORT_MEDIA_MODES = (
+    MediaPlayerEntityFeature.PAUSE
+    | MediaPlayerEntityFeature.STOP
+    | MediaPlayerEntityFeature.PREVIOUS_TRACK
+    | MediaPlayerEntityFeature.NEXT_TRACK
+    | MediaPlayerEntityFeature.PLAY
+    | MediaPlayerEntityFeature.SEEK
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up the Devialet entry."""
+    client = hass.data[DOMAIN][entry.entry_id]
+
+    async_add_entities(
+        [DevialetDevice(client, entry)],
+        update_before_add=True,
+    )
+
+
+class DevialetDevice(MediaPlayerEntity):
+    """Representation of a Devialet device."""
+
+    def __init__(self, client: DevialetApi, entry: ConfigEntry) -> None:
+        """Initialize the Devialet device."""
+        self._client = client
+        self._name = entry.data[CONF_NAME]
+        self._muted = False
+        if entry.unique_id:
+            self._serial = entry.unique_id
+
+    async def async_update(self) -> None:
+        """Get the latest details from the device."""
+        await self._client.async_update()
+
+    @property
+    def device_info(self) -> DeviceInfo:
+        """Return the device info."""
+        return DeviceInfo(
+            identifiers={(DOMAIN, self._serial)},
+            name=self._name,
+            manufacturer=MANUFACTURER,
+            model=self._client.model,
+            sw_version=self._client.version,
+        )
+
+    @property
+    def name(self) -> str:
+        """Return the name of the device."""
+        return self._name
+
+    @property
+    def unique_id(self) -> str:
+        """Return the unique id of the device."""
+        return self._serial
+
+    @property
+    def state(self) -> MediaPlayerState | None:
+        """Return the state of the device."""
+        if not self._client.is_available:
+            return MediaPlayerState.OFF
```

Comment:
If the device is offline/unreachable we should set the `available` property to false

Reply:
As documented, the Devialet speakers have a physical off-switch. When using this switch, the device should not go to unavailable as this stops updating the state and requires a manual reload of the integration to get it active again..

Reply:
It seems, this is fixed now, nice 👍